site/app/Http/Controllers/TagController.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Course;
+use App\Http\Requests\IndexTag;
+use App\Http\Requests\StoreTag;
+use App\Http\Requests\UpdateTag;
+use App\Tag;
+use Illuminate\Support\Facades\Auth;
+
+class TagController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return Renderable
+     *
+     * @throws AuthorizationException
+     */
+    public function index(Course $course, IndexTag $request)
+    {
+        $this->authorize('viewInCourseConfiguration', [Tag::class, $course]);
+
+        $tagOrder = $request->get('tag_order') ?? 'name';
+        $tagDirection = $request->get('tag_direction') ?? 'asc';
+
+        $tags = Tag::where('course_id', $course->id)
+            ->selectRaw('tags.id, course_id, name, count(card_tag.tag_id) as cards_count')
+            ->leftJoin('card_tag', 'tags.id', '=', 'card_tag.tag_id')
+            ->groupBy('tags.id', 'course_id', 'name')
+            ->orderBy($tagOrder, $tagDirection)
+            ->orderBy('name', 'asc')
+            ->orderBy('cards_count', 'asc')
+            ->get();
+
+        // Get inversed order values for each columns (for url generation).
+        $tagColumns = array_merge(
```

Comment:
We need data from the request ($tagOrder and $tagDirection) to generate these values. The data are not available in the view and should be initialized again. We could pass these data to the view but it would make the code harder to understand that it should be.

```php
  $tagOrder = $request->get('tag_order') ?? 'name';
  $tagDirection = $request->get('tag_direction') ?? 'asc';
```